packages/metro/src/Server.js

```diff
@@ -1153,6 +1153,55 @@ class Server {
         prepend = [];
       }
 
+      // If the project defines a custom serializer, then we should run it to ensure any
+      // mutations to the graph are applied in the sourcemap.
```

Comment:
It makes sense to me to run the custom serializer consistently (and I vaguely remember seeing cases where this was broken), but what do you mean by "mutations to the graph"? When during bundling do those occur?

Reply:
If someone were to `preModules.splice(0,0, { ... })` to add a new module inside of a custom serializer, then it would technically mutate the arguments used to generate the bundle. I can think of two cases like this:
- [sentry](https://github.com/getsentry/sentry-react-native/blob/0019f64e4d91eddd0c5c8cdc369778caecc4815c/src/js/tools/sentryMetroSerializer.ts#L126-L138) (not exactly, but close) 
- One I just wrote for Expo CLI as a fallback to returning the map: [here](https://github.com/expo/expo/pull/29463/files#diff-fd86a91bdc3cee21b84074550e6ec3a5651e6c6ca2b3dd9b9eccfa9e1b13aa2aR100)

Reply:
Hmm. The serializer API wasn't designed with side-effectful serializers in mind, and the `preModules` parameter specifically is typed as being read-only. Is there a way to achieve what you're trying to do without mutating read-only arguments?
 
<img width="761" alt="image" src="https://github.com/facebook/metro/assets/2246565/9adc704c-65dc-4247-96f3-f6f814f82e65">

Reply:
Yeah but it's a little unstable. I'm sniffing if the sourceUrl has a `.map` pathname to determine if the serializer should return source maps. This works for the `.map` endpoint, but it doesn't account for `/symbolicate` where the result is `getExplodedSourceMap` and not a standard source map string.